### PR TITLE
Backend api

### PR DESCRIPTION
**Pull request by vaishnaviyathirajam**
Github Issue: #9

### Description

Updated code that works for multiple layers

### CODE REVIEW

**Comment by frankinspace** [comment]
Can you remove the files that are no longer relevant?

File: dag/api.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+# CYGNSS_L3_CDR_V1.1
+# GHRSST_L4_MUR_Sea_Surface_Temperature
+# GHRSST_L4_MUR25_Sea_Ice_Concentration
+# http://ec2-3-65-18-201.eu-central-1.compute.amazonaws.com:8080/get_stats?minx=-160&miny=-5&maxx=-150&maxy=0&timestamp=2016-06-09T00:00:00Z&format=json
+@app.get("/get_stats", tags=["Home"])
+def get_stats(minx:float,miny:float,maxx:float,maxy:float,timestamp:str,format:str, layer='GHRSST_L4_MUR_Sea_Surface_Temperature', colormap = 'GHRSST_Sea_Surface_Temperature.xml', _scale=1):
+    
+    import subprocess
+    return  json.loads(subprocess.check_output(['python', 'get_stats.py', str(minx), str(miny), str(maxx), str(maxy), str(timestamp), str(_scale), str(layer), str(colormap)]).decode())
```
Comment (frankinspace) [inline dag/api.py:39]: Why are we starting subprocesses?

Comment (vaishnaviyathirajam) [reply dag/api.py:39]: In order to serve multiple requests at a time. I ma creating a sub process for each request

Comment (vaishnaviyathirajam) [reply dag/api.py:39]: Also, numpy was throwing an error when run under uvicorn, having to do with the null values. As subprocess runs under vanilla python, its not affected by any issues created by using uvicorn

Comment (frankinspace) [reply dag/api.py:39]: uvicorn is responsible for request handling. At this point the request is already in it's own thread/process we don't need to start another one.

**Comment by frankinspace** [comment]
There's still a lot of repeated code here; which one is the version that's actually getting run? Can you remove the code that is not being used.

File: dag/api.py (hunk 1 above)
Comment (vaishnaviyathirajam) [reply dag/api.py:39]: > There's still a lot of repeated code here; which one is the version that's actually getting run? Can you remove the code that is not being used.



> There's still a lot of repeated code here; which one is the version that's actually getting run? Can you remove the code that is not being used.

Ok. I am basically working only in this branch. Should i delete all the files in the other branch and the ones that i am not using here as well?

**Comment by frankinspace** [comment]
@vaishnaviyathirajam We should get this merged to develop before deploying to any environments; can you address the remaining comments?

**Comment by vaishnaviyathirajam** [comment]
Ok Frank. I will upload the updated code I am working on currently and delete all the others by tomorrow. Will update you once that's done.

**Comment by vaishnaviyathirajam** [comment]
Frank, can you please review and check if its good to be merged to develop

**Comment by vaishnaviyathirajam** [comment]
> There's still a lot of repeated code here; which one is the version that's actually getting run? Can you remove the code that is not being used.

Done

File: dag/api.py (hunk 1 above)
Comment (frankinspace) [reply dag/api.py:39]: We can address the subprocess in https://github.com/podaac/gibs-imagestat/issues/11

**Review by frankinspace — APPROVED** [review]

**Pull request merged**